src/chunk_constraint.c

```diff
@@ -444,6 +444,63 @@ typedef struct ChunkConstraintScanData
 	DimensionSlice *slice;
 } ChunkConstraintScanData;
 
+/*
+ * Scan for all chunk constraints that match the given slice ID. The chunk
+ * constraints are not saved, only the matching chunk id is returned.
+ */
+int
+ts_chunk_constraint_scan_by_dimension_slice_chunk_id(const DimensionSlice *slice, ChunkScanCtx *ctx,
+													 MemoryContext mctx)
+{
+	int result = 0;
+
+	ScanIterator iterator = ts_scan_iterator_create(CHUNK_CONSTRAINT, AccessShareLock, mctx);
+	ts_chunk_constraint_scan_iterator_set_slice_id(&iterator, slice->fd.id);
+
+	ts_scanner_foreach(&iterator)
+	{
+		ChunkScanEntry *entry;
+		bool found;
+		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
+		Datum datum = slot_getattr(ti->slot, Anum_chunk_constraint_chunk_id, &found);
+		int32 chunk_id = DatumGetInt32(datum);
+
+		if (slot_attisnull(ts_scan_iterator_slot(&iterator),
+						   Anum_chunk_constraint_dimension_slice_id))
+			continue;
+
+		entry = hash_search(ctx->htab, &chunk_id, HASH_ENTER, &found);
+
+		if (!found)
+		{
+			entry->stub = NULL;
+			entry->num_dimension_constraints = 0;
+		}
+
+		/*
+		 * We have only the dimension constraints here, because we're searching
+		 * by dimension slice id.
+		 */
+		Assert(!slot_attisnull(ti->slot, Anum_chunk_constraint_dimension_slice_id));
```

Comment:
Can remove this assert since there is already a null check for this dimension slice ID above.

Reply:
I removed the null check instead.